tree/ntuple/v7/src/RField.cxx

```diff
@@ -772,26 +772,27 @@ ROOT::Experimental::RFieldBase::Create(const std::string &fieldName, const std::
          return R__FORWARD_RESULT(fnFail("the type list for std::map must have exactly two elements"));
       }
 
-      auto normalizedKeyTypeName = GetNormalizedTypeName(innerTypes[0]);
-      auto normalizedValueTypeName = GetNormalizedTypeName(innerTypes[1]);
+      // We need to create fields for the key and value in order to get the correct (normalized) type name. The fields
+      // themselves won't be used.
+      auto keyTypeName = Create("", innerTypes[0]).Unwrap()->GetTypeName();
```

Comment:
This seems odd.  Can you clarify why `GetNormalizedTypeName` is not recursively normalizing the names?

Reply:
I don't know why it's not done recursively -- my guess would be that it is assumed that fields are always created recursively anyways so the type names get resolved that way, which is the case for all field types except `std::map` and friends. It is different for `std::map` because rather than creating subfields for the key and value, we create a `std::pair` subfield.  If my memory serves correctly, we decided to do it this way because this gave compatibility with the way (associative) collection proxies handle things. 

`GetNormalizedTypeName` uses the `typeTranslationMap` for normalizing fundamental types, which before was an exhaustive mapping, but now much of this is handled by `RIntegralTypeMap`.

Reply:
My biggest problem with this is that now  'GetNormalizedTypeName` does not live up to it names (i.e. acts as if it is something like 'GetSomewhatNormalizedTypeName` :) ) and will be hard to remember/keep track when it is applicable and when it is not.  Doing the recursive normalization is straight-forward (but indeed not trivial: split the name in template name and set of template arguments, then recurse and then reassemble;  This is implemented in Core/Meta version of the name normalization routine.

Reply:
I agree that it would be good if it was recursive, but that won't solve the issue addressed in this PR. The issue in this case is that integer key/value type names of a map aren't properly converted to their fixed-width type name without actually turning them into a proper field first, because the responsibility of assigning the fixed-width type name has been moved from `GetNormalizedTypeName` (via [`typeTranslationMap`](https://github.com/root-project/root/blob/1287aaed08137075693b1c0c99fe6c02cf296e56/tree/ntuple/v7/src/RField.cxx#L61C1-L92)) to the template specializations of `RIntegralTypeMap` (see #16039). If we want to do the full, string-only based recursive type name normalization it would mean we would have to extend the `typeTranslationMap` again to include every possible typedef/fundamental type name.

Reply:
> I agree that it would be good if it was recursive, 

Humm ... it sounds like it would actually be a fatal flaw in the function (i.e. the returned name would *not* be the normalized name but some hybrid).

> but that won't solve the issue addressed in this PR ... 

Fair enough but that raises a related question, what if the user is attempting to create a column for an std pair and passes a non-normalized name, would that work (and why?) or would that fail in the same way the original code in this PR was failing?